src/model/channel/attachment.rs

```diff
@@ -30,6 +30,12 @@ pub struct Attachment {
     ///
     /// [media type]: https://en.wikipedia.org/wiki/Media_type
     pub content_type: Option<String>,
+    /// Whether this attachment is ephemeral.
+    ///
+    /// Ephemeral attachments will automatically be removed after a set period of time.
+    /// Ephemeral attachments on messages are guaranteed to be available as long as
+    /// the message itself exists.
+    pub ephemeral: Option<bool>,
```

Comment:
I think this should be a regular `bool` with the `#[serde(default)]` attribute. It is not clear based on the documentation about the state of the field, which has the potential to adopt one of the 3 states:
1. `None` -- unsure what this is supposed to represent, or when it may be realised; is it when the attachment isn't ephemeral, or...?
2. `Some(false)` -- the attachment is not ephemeral (or something else, 1. muddles this state)
3. `Some(true)` -- the attachment is ephemeral

```suggestion
    #[serde(default)]
    pub ephemeral: bool,
```

Furthermore, it might be wise to skip serialising the field if it is `false`:
```suggestion
    #[serde(default, skip_serializing_if = "is_false")]
    pub ephemeral: bool,
```
```rs
fn is_false(b: &bool) -> bool {
    !b
}
```